xbout/boutdataset.py

```diff
@@ -239,6 +242,50 @@ def to_restart(self, savepath='.', nxpe=None, nype=None,
             save_mfdataset(restart_datasets, paths, compute=True)
         return
 
+    def animate_list(self, variables, animate_over='t', save_as=None, show=False, fps=10,
+                     nrows=None, ncols=None, **kwargs):
+        nvars = len(variables)
+
+        if nrows is None and ncols is None:
+            ncols = int(np.ceil(np.sqrt(nvars)))
+            nrows = int(np.ceil(nvars/ncols))
+        elif nrows is None:
+            nrows = int(np.ceil(nvars/ncols))
+        elif ncols is None:
+            ncols = int(np.ceil(nvars/nrows))
+        else:
+            assert nrows*ncols >= nvars, 'nrows*ncols not as big as nvar'
```

Comment:
Asserts [shouldn't be used for error/warning checking](https://stackoverflow.com/questions/944592/best-practice-for-python-assert) (or for any flow control in general), because they will be [ignored](https://stackoverflow.com/questions/2830358/what-are-the-implications-of-running-python-with-the-optimize-flag.
) if you run python with the `-O` or `-OO` optimisation flags.

This should instead be
```python
if nrows * ncols < nvars:
    raise ValueError("Not enough rows * columns to fit all variables")
```